byteparse/shared/src/main/scala/byteparse/classparse/ClassAttributes.scala

```diff
@@ -0,0 +1,183 @@
+package byteparse
+package classparse
+
+import fastparse.allByte._
+
+object ClassAttributes {
+  import ClassParser.Info._
+  import ClassParser.BasicElems._
+  import ClassParser.Ast._
+  import ClassParser._
+  import CodeParser._
+
+  sealed abstract class Attribute
+
+  case class BasicAttribute(name: String, info: Array[Byte]) extends Attribute {
+    override def equals(other: Any): Boolean =
+      other match {
+        case attr: BasicAttribute => name == attr.name && info.deep == attr.info.deep
+        case _ => false
+      }
+
+    override def toString = {
+      import fastparse.ElemTypeFormatter.ByteFormatter
+      s"BasicAttribute($name,${ByteFormatter.prettyPrint(info)})"
+    }
+  }
+
+  case class ConstantValueAttribute(value: BasicElem) extends Attribute
+
+  case class ExceptionHandler(startIdx: Int, endIdx: Int, handlerIdx: Int, catchType: Option[Class])
+
+  case class CodeAttribute(maxStack: Int, maxLocals: Int,
+                           code: Seq[OpCode], exceptions: Seq[ExceptionHandler],
+                           attributes: Seq[Attribute]) extends Attribute
+
+  case class ExceptionsAttribute(exceptions: Seq[Class]) extends Attribute
+
+
+  case class InnerClass(innerClass: Class, outerClass: Option[Class],
+                        innerName: Option[String], accessFlags: InnerClassFlags)
+
+  case class InnerClassesAttribute(classes: Seq[InnerClass]) extends Attribute
+
+  case class EnclosingMethodAttribute(enclosingClass: Class, enclosingMethodName: Option[String],
+                                      enclosingMethodDescriptor: Option[String]) extends Attribute
+
+  case class SyntheticAttribute() extends Attribute
+
+  case class SignatureAttribute(signature: String) extends Attribute
+
+  case class SourceFileAttribute(sourceFile: String) extends Attribute
+
+  case class DeprecatedAttribute() extends Attribute
+
+  case class BootstrapMethod(bootstrapMethodRef: MethodHandle, bootstrapArguments: Seq[PoolItem])
+
+  case class BootstrapMethodsAttribute(bootstrapMethods: Seq[BootstrapMethod]) extends Attribute
+
+  import ByteUtils.BE._
+
+  val attributeParsers = Map[String, (ClassFileInfo) => Parser[Attribute]](
```

Comment:
Also, the `Map` here seems big enough it's worth lifting each `ClassFileInfo => Attribute` into a standalone method/factory, and then referring to it in the `Map`. As it stands now the `Map` is kind of sprawling and difficult to skim through, e.g. if we forgot one case it would not be obvious